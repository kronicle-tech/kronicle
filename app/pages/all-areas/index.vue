<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllAreasTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click an area's name in the table below to view more information about
            that area
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="area-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="area in areas" :key="area.id">
                <td class="area-name table-primary">
                  <AreaName :area="area" />
                </td>
                <td>
                  <Markdown :markdown="area.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Area } from '~/types/kronicle-service'
import AllAreasTabs from '~/components/AllAreasTabs.vue'
import AreaName from '~/components/AreaName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllAreasTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    AreaName,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id,name,description)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    return {
      areas,
    }
  },
  data() {
    return {
      areas: [] as Area[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Areas',
    }
  },
})
</script>
