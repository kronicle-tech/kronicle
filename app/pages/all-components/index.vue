<template>
  <div>
    <AllComponentsTabs />
    <b-alert show="10" dismissible variant="info" class="my-3">
      Click a component's name in the table below to view more information about
      that component
    </b-alert>

    <ComponentsView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component } from '~/types/kronicle-service'
import AllComponentsTabs from '~/components/AllComponentsTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AllComponentsTabs,
    'b-alert': BAlert,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    store.commit('componentFilters/initialize', {
      components,
      route,
    })

    return {
      components,
    }
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Components',
    }
  },
})
</script>
