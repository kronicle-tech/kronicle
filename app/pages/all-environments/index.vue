<template>
  <div class="m-3">
    <EnvironmentStatesView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {NuxtRuntimeConfig} from "@nuxt/types/config/runtime";
import {Route} from "vue-router";
import {Store} from "vuex";
import EnvironmentStatesView from '~/components/EnvironmentStatesView.vue'
import { Component } from '~/types/kronicle-service'

interface Data {
  components: Component[]
}

async function loadData(
  { $config, route, store }: { $config: NuxtRuntimeConfig, route: Route, store: Store<any> }
): Promise<Data> {
  const components = await fetch(
    `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId,states)`
  )
    .then((res) => res.json())
    .then((json) => json.components as Component[])

  store.commit('componentFilters/initialize', {
    components,
    route,
  })

  return {
    components: [] as Component[],
  }
}

export default Vue.extend({
  components: {
    EnvironmentStatesView,
  },
  async asyncData({ $config, route, store }) {
    return await loadData({
      $config,
      route,
      store,
    })
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Environments',
    }
  },
  mounted() {
    setInterval(
      async () => {
        const data = await loadData({
          $config: this.$config,
          route: this.$route,
          store: this.$store
        })
        this.components = data.components
      },
      60 * 1000
    )
  },
})
</script>
