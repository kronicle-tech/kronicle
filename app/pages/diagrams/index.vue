<template>
  <div class="m-3">
    <DiagramsView :diagrams="diagrams" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Diagram } from '~/types/kronicle-service'
import DiagramsView from '~/components/DiagramsView.vue'

export default Vue.extend({
  components: {
    DiagramsView,
  },
  async asyncData({ $config }) {
    const diagrams = await fetch(
      `${$config.serviceBaseUrl}/v1/diagrams?fields=diagrams(id,name,type,tags)`
    )
      .then((res) => res.json())
      .then((json) => json.diagrams)

    return {
      diagrams,
    }
  },
  data() {
    return {
      diagrams: [] as Diagram[],
    }
  },
  head() {
    return {
      title: 'Kronicle - Diagrams',
    }
  },
})
</script>
