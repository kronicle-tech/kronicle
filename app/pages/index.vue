<template>
  <div class="m-3">
    <Intro id="intro" :title="introTitle" :markdown="introMarkdown" />

    <b-card-group columns>
      <b-card
        header="Kronicle"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Kronicle contains the following:
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-areas' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Areas</b>
            <span class="lead">
              <b-badge class="area-count" variant="primary" pill>{{
                  areaCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-teams' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Teams</b>
            <span class="lead">
              <b-badge class="team-count" variant="primary" pill>{{
                  teamCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Components</b>
            <span class="lead">
              <b-badge class="component-count" variant="primary" pill>{{
                  componentCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tech-debts' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Tech Debts</b>
            <span class="lead">
              <b-badge class="tech-debt-count" variant="primary" pill>{{
                  techDebtCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tests' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Test Results</b>
            <span class="lead">
              <b-badge class="test-result-count" variant="primary" pill>{{
                  testResultCount
                }}</b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Scanners"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Scanners dynamically add data to Kronicle at runtime:
          </b-list-group-item>

          <b-list-group-item v-for="scanner in scanners" :key="scanner.id">
            <b class="scanner-id text-info">{{ scanner.id }}</b>
            <br />
            <Markdown :markdown="scanner.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tests"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Tests evaluate the data associated with components in Kronicle and
            return outcomes of pass, fail and not applicable
          </b-list-group-item>

          <b-list-group-item
            v-for="test in tests"
            :key="test.id"
            :to="{
              name: 'tests-testId',
              params: { testId: test.id },
            }"
          >
            <b class="test-id text-info">{{ test.id }}</b>
            <br />
            <Markdown :markdown="test.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Platforms"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group-item
          class="d-flex justify-content-between align-items-center text-muted"
        >
          The following platforms are associated with components in Kronicle:
        </b-list-group-item>

        <b-list-group flush>
          <b-list-group-item
            v-for="platformCount in platformCounts"
            :key="platformCount.item"
            :to="{
              name: 'all-components',
              query: { platformId: platformCount.item || 'undefined' },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="platform">{{ platformCount.item || 'missing' }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ platformCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tags"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            The components in Kronicle are tagged with the following:
          </b-list-group-item>

          <b-list-group-item
            v-for="tagCount in tagCounts"
            :key="tagCount.item"
            :to="{
              name: 'all-components',
              query: { tag: tagCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="tag">{{ tagCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ tagCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Component Types"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            There are the following types of components in Kronicle:
          </b-list-group-item>

          <b-list-group-item
            v-for="componentTypeCount in componentTypeCounts"
            :key="componentTypeCount.item"
            :to="{
              name: 'all-components',
              query: { componentType: componentTypeCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="component-type">{{ componentTypeCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ componentTypeCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>
    </b-card-group>
  </div>
</template>

<style scoped>
.description {
  display: inline-block;
  color: white;
  text-decoration: none;
}

.description >>> p {
  margin-bottom: 0;
}
</style>

<script lang="ts">
import Vue from 'vue'
import {
  BBadge,
  BCard,
  BCardGroup,
  BListGroup,
  BListGroupItem,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Area, Component, Scanner, Team, Test } from '~/types/kronicle-service'
import { ItemCount, itemCounts } from '~/src/arrayUtils'
import Intro from '~/components/Intro.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    'b-card': BCard,
    'b-card-group': BCardGroup,
    'b-list-group': BListGroup,
    'b-list-group-item': BListGroupItem,
    Intro,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,typeId,platformId,tags,techDebts(doesNotExist),testResults(doesNotExist))`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    const tests = await fetch(
      `${$config.serviceBaseUrl}/v1/tests?fields=tests(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.tests as Test[])

    return {
      areas,
      teams,
      components,
      scanners,
      tests,
    }
  },
  data() {
    return {
      introTitle: this.$config.introTitle as String | undefined,
      introMarkdown: this.$config.introMarkdown as String | undefined,
      areas: [] as Area[],
      teams: [] as Team[],
      components: [] as Component[],
      scanners: [] as Scanner[],
      tests: [] as Test[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle',
    }
  },
  computed: {
    areaCount(): number {
      return this.areas.length
    },
    teamCount(): number {
      return this.teams.length
    },
    componentCount(): number {
      return this.components.length
    },
    componentTypeCounts(): ItemCount<string>[] {
      return itemCounts(this.components.map((component) => component.typeId))
    },
    platformCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.map((component) => component.platformId)
      )
    },
    tagCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.flatMap((component) => component.tags ?? [])
      )
    },
    techDebtCount(): number {
      return this.components.flatMap((component) => component.techDebts ?? [])
        .length
    },
    testResultCount(): number {
      return this.components.flatMap((component) => component.testResults ?? [])
        .length
    },
  },
})
</script>
