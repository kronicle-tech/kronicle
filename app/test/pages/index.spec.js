import Index from '@/pages/index.vue'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createArea,
  createComponent,
  createScanner,
  createTeam,
  createTest,
} from '~/test/testDataUtils'
import { expectTextsInItems } from '~/test/pages/wrapperUtils'

describe('Index', () => {
  let areas
  let teams
  let components
  let scanners
  let tests
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      serviceRequests: {
        '/v1/areas?fields=areas(id)': {
          responseBody: { areas },
        },
        '/v1/teams?fields=teams(id)': {
          responseBody: { teams },
        },
        '/v1/components?fields=components(id,typeId,platformId,tags,techDebts(doesNotExist),testResults(doesNotExist))':
          {
            responseBody: { components },
          },
        '/v1/scanners?fields=scanners(id,description)': {
          responseBody: { scanners },
        },
        '/v1/tests?fields=tests(id,description)': {
          responseBody: { tests },
        },
      },
    })
  }

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when the catalog contains no information', () => {
    beforeEach(() => {
      areas = []
      teams = []
      components = []
      scanners = []
      tests = []
    })

    test('renders the page with no information on it', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.get('.area-count').text()).toEqual('0')
      expect(wrapper.get('.team-count').text()).toEqual('0')
      expect(wrapper.get('.component-count').text()).toEqual('0')
      expect(wrapper.get('.tech-debt-count').text()).toEqual('0')
      expect(wrapper.get('.test-result-count').text()).toEqual('0')
    })
  })

  describe('when the catalog contains information', () => {
    beforeEach(() => {
      areas = [
        createArea({ areaNumber: 1 }),
        createArea({ areaNumber: 2 }),
        createArea({ areaNumber: 3 }),
      ]
      teams = [
        createTeam({ teamNumber: 1 }),
        createTeam({ teamNumber: 2 }),
        createTeam({ teamNumber: 3 }),
      ]
      components = [
        createComponent({
          componentNumber: 1,
          platformNumber: 1,
          hasTechDebts: true,
          hasTestResults: true,
        }),
        createComponent({
          componentNumber: 2,
          hasTechDebts: true,
          hasTestResults: true,
        }),
        createComponent({
          componentNumber: 3,
          platformNumber: 3,
          hasTechDebts: true,
          hasTestResults: true,
        }),
      ]
      scanners = [
        createScanner({ scannerNumber: 1 }),
        createScanner({ scannerNumber: 2 }),
        createScanner({ scannerNumber: 3 }),
      ]
      tests = [
        createTest({ testNumber: 1 }),
        createTest({ testNumber: 2 }),
        createTest({ testNumber: 3 }),
      ]
    })

    test('has the right page title', async () => {
      await createWrapper()
      expect(wrapper.vm.$metaInfo.title).toBe('Component Catalog')
    })

    test('renders the page showing the information', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.get('.area-count').text()).toEqual('3')
      expect(wrapper.get('.team-count').text()).toEqual('3')
      expect(wrapper.get('.component-count').text()).toEqual('3')
      expect(wrapper.get('.tech-debt-count').text()).toEqual('6')
      expect(wrapper.get('.test-result-count').text()).toEqual('6')
      expectTextsInItems(wrapper, '.scanner-id', [
        'test-scanner-id-1',
        'test-scanner-id-2',
        'test-scanner-id-3',
      ])
      expectTextsInItems(wrapper, '.test-id', [
        'test-test-id-1',
        'test-test-id-2',
        'test-test-id-3',
      ])
      expectTextsInItems(wrapper, '.platform', [
        'missing',
        'test-platform-id-1',
        'test-platform-id-3',
      ])
      expectTextsInItems(wrapper, '.tag', [
        'test-tag-1-1',
        'test-tag-1-2',
        'test-tag-2-1',
        'test-tag-2-2',
        'test-tag-3-1',
        'test-tag-3-2',
      ])
      expectTextsInItems(wrapper, '.component-type', [
        'test-component-type-id-1',
        'test-component-type-id-2',
        'test-component-type-id-3',
      ])
    })
  })
})
