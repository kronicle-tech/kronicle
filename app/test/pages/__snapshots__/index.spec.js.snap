// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Kronicle contains information renders the page showing the information 1`] = `
<div
  class="m-3"
>
  <div
    class="jumbotron my-3"
    id="intro"
  >
    <h1
      class="display-3"
    >
      Test Intro Title
    </h1>
    <!---->
    <div>
      

      <p>
        Test Intro 
        <em>
          Markdown
        </em>
      </p>
    </div>
  </div>
   
  <div
    class="card-columns"
  >
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Kronicle
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Kronicle contains the following:
        
        </div>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Areas
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge area-count badge-primary badge-pill"
            >
              3
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Teams
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge team-count badge-primary badge-pill"
            >
              3
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Components
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge component-count badge-primary badge-pill"
            >
              3
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Tech Debts
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge tech-debt-count badge-primary badge-pill"
            >
              6
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Test Results
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge test-result-count badge-primary badge-pill"
            >
              6
            </span>
          </span>
        </a>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Scanners
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Scanners dynamically add data to Kronicle at runtime:
        
        </div>
         
        <div
          class="list-group-item"
        >
          <b
            class="scanner-id text-info"
          >
            test-scanner-id-1
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Scanner Description 1
            </p>
          </div>
        </div>
        <div
          class="list-group-item"
        >
          <b
            class="scanner-id text-info"
          >
            test-scanner-id-2
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Scanner Description 2
            </p>
          </div>
        </div>
        <div
          class="list-group-item"
        >
          <b
            class="scanner-id text-info"
          >
            test-scanner-id-3
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Scanner Description 3
            </p>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Tests
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Tests evaluate the data associated with components in Kronicle and
          return outcomes of pass, fail and not applicable
        
        </div>
         
        <a
          class="list-group-item list-group-item-action"
          href="#"
          target="_self"
        >
          <b
            class="test-id text-info"
          >
            test-test-id-1
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Test Description 1
            </p>
          </div>
        </a>
        <a
          class="list-group-item list-group-item-action"
          href="#"
          target="_self"
        >
          <b
            class="test-id text-info"
          >
            test-test-id-2
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Test Description 2
            </p>
          </div>
        </a>
        <a
          class="list-group-item list-group-item-action"
          href="#"
          target="_self"
        >
          <b
            class="test-id text-info"
          >
            test-test-id-3
          </b>
           
          <br />
           
          <div
            class="description"
          >
            

            <p>
              Test Test Description 3
            </p>
          </div>
        </a>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Platforms
      </div>
      <div
        class="list-group-item d-flex justify-content-between align-items-center text-muted"
      >
        
        The following platforms are associated with components in Kronicle:
      
      </div>
       
      <div
        class="list-group list-group-flush"
      >
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?platformId=undefined"
          target="_self"
        >
          <b
            class="platform"
          >
            missing
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?platformId=test-platform-id-1"
          target="_self"
        >
          <b
            class="platform"
          >
            test-platform-id-1
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?platformId=test-platform-id-3"
          target="_self"
        >
          <b
            class="platform"
          >
            test-platform-id-3
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Tags
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          The components in Kronicle are tagged with the following:
        
        </div>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-1-1"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-1-1
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-1-2"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-1-2
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-2-1"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-2-1
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-2-2"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-2-2
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-3-1"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-3-1
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?tag=test-tag-3-2"
          target="_self"
        >
          <b
            class="tag"
          >
            test-tag-3-2
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Component Types
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          There are the following types of components in Kronicle:
        
        </div>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?componentType=test-component-type-id-1"
          target="_self"
        >
          <b
            class="component-type"
          >
            test-component-type-id-1
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?componentType=test-component-type-id-2"
          target="_self"
        >
          <b
            class="component-type"
          >
            test-component-type-id-2
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="?componentType=test-component-type-id-3"
          target="_self"
        >
          <b
            class="component-type"
          >
            test-component-type-id-3
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge badge-primary badge-pill"
            >
              
              1
            
            </span>
          </span>
        </a>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`Index when Kronicle contains no information renders the page with no information on it 1`] = `
<div
  class="m-3"
>
  <div
    class="jumbotron my-3"
    id="intro"
  >
    <h1
      class="display-3"
    >
      Test Intro Title
    </h1>
    <!---->
    <div>
      

      <p>
        Test Intro 
        <em>
          Markdown
        </em>
      </p>
    </div>
  </div>
   
  <div
    class="card-columns"
  >
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Kronicle
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Kronicle contains the following:
        
        </div>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Areas
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge area-count badge-primary badge-pill"
            >
              0
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Teams
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge team-count badge-primary badge-pill"
            >
              0
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Components
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge component-count badge-primary badge-pill"
            >
              0
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Tech Debts
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge tech-debt-count badge-primary badge-pill"
            >
              0
            </span>
          </span>
        </a>
         
        <a
          class="list-group-item d-flex justify-content-between align-items-center list-group-item-action"
          href="#"
          target="_self"
        >
          <b>
            Test Results
          </b>
           
          <span
            class="lead"
          >
            <span
              class="badge test-result-count badge-primary badge-pill"
            >
              0
            </span>
          </span>
        </a>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Scanners
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Scanners dynamically add data to Kronicle at runtime:
        
        </div>
         
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Tests
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          Tests evaluate the data associated with components in Kronicle and
          return outcomes of pass, fail and not applicable
        
        </div>
         
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Platforms
      </div>
      <div
        class="list-group-item d-flex justify-content-between align-items-center text-muted"
      >
        
        The following platforms are associated with components in Kronicle:
      
      </div>
       
      <div
        class="list-group list-group-flush"
      />
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Tags
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          The components in Kronicle are tagged with the following:
        
        </div>
         
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="card"
    >
      <!---->
      <div
        class="card-header lead bg-info"
      >
        Component Types
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="list-group-item d-flex justify-content-between align-items-center text-muted"
        >
          
          There are the following types of components in Kronicle:
        
        </div>
         
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
